remote/remote_state_s3.go

```diff
@@ -123,6 +130,28 @@ func (s3Config *RemoteStateConfigS3) GetLockTableName() string {
 	return s3Config.LockTable
 }
 
+// Since Terraform 1.6 assume_role is a new block but we need to keep deprecated top level arguments
```

Comment:
```
Comment should have the following format 'GetSessionRoleArn ...' (with an optional leading article) 
```

Reply:
I fixed the comment but maybe it would be worth removing it.

Reply:
I will raise a separate PR to clean the comments